### ISSUE TITLE
User Unable to Whitelist when Admin Password Blank

### ISSUE THREAD

**Issue by recrudesce** [pi-hole/pi-hole#1670]
- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] The issue I am reporting can be *replicated*
- [X] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the codebase?:**

0
---
**[OTHER]:**

**[ISSUE] Expected Behaviour:**
I run a Pihole at home, I'm the only one who uses it, and the only one who has access to it.  I have, therefore, removed the admin password.  If I go to a website that is blacklisted and get presented with the "Website Blocked" page, I should be able to add things to the whitelist.

**[ISSUE] Actual Behaviour:**
I cannot whitelist through the "Website Blocked" page presented.  I get `---> 
No password set - whitelisting with password not supported <---`

**[ISSUE] Steps to reproduce:**
Default installation of Pihole, remove the password, implement the following lists
```
list.0.raw.githubusercontent.com.domains 
list.preEventHorizon 
```

go to a bit.ly or owl.ly link to get the Website Blocked page, click the Whitelist this page button, and try and whitelist with an empty password.

**Comment by recrudesce** [comment]
I've been thinking about this... considering you give us a way to disable the password through command line (ie requiring some level of admin access to the box to achieve it), could we have a "allow whitelisting without password" option in a config file, which subsequently will also only be editable by someone with the required permissions ?

**Comment by Mcat12** [comment]
You can add a feature request on [Discourse](https://discourse.pi-hole.net/).

**Comment by PromoFaux** [comment]
That's actually not a bad idea, it's essentially a "I understand the risks I am exposing myself to, and understand Pi-hole cannot be held accountable if the internet kills my dog" option.

What was the main reason we added the password requirement to the blockpage in the first place?

**Comment by Mcat12** [comment]
It requires a password to whitelist a domain (assuming you stick with the default of having a password)

**Comment by PromoFaux** [comment]
But if you have no password on the interface, you can still whitelist a page from the interface, right?

**Comment by Mcat12** [comment]
From the web interface, yes, but we didn't account for that case in the block page (it assumes there's a password)

**Comment by frodehb** [comment]
This.

**Comment by Mcat12** [comment]
Will be fixed by pi-hole/AdminLTE#597

**Comment by jacobsalmela** [comment]
Resolved in v3.2 ( pi-hole/AdminLTE#597)
